### PR TITLE
Fix PDF Export bug

### LINKED ISSUE #715: Bug Report - Incorrect Values in PDF Export from Current View
Description:
I would like to report a bug regarding the PDF export feature in our application. When exporting the current view to PDF format, the overview values (total expenses and total income) do not reflect the data from the current view; instead, they display the values from the whole account.

Expected Behavior:
The expected behaviour is that when exporting the current view to PDF, the overview values (total expenses and total income) should accurately represent the data present in the current view.

Steps to Reproduce:

1. Create a view, for instance, by clicking on the button labelled **Select current month**.
2. Click on the "Actions" menu and select the "Export as PDF" option.
3. Choose the "Export current view" option in the PDF export dialogue.
4. Observe that the generated PDF contains incorrect values for total expenses and total income, which should match the data in the current view.

Version: 2023.9.2

### PR DESCRIPTION
**Pull request by JoseBritto**
Fixes #715 
Replaces #716  

- Added GetGroupIncome,Expense and Total methods.
- Simplified existing GetIncome,Expense and Total methods

@nlogozzo Let me know if anything needs to be changed.

### CODE REVIEW

**Comment by nlogozzo** [comment]
Could you update the changelog in `GNOME/Program.cs` and `Shared/org.nickvision.money.metainfo.xml`? 

Something like `- Fixed an issue where exported PDF values were incorrect` is fine.

File: NickvisionMoney.Shared/Models/Account.cs [hunk 1]
```diff
@@ -550,90 +550,94 @@ public async Task<bool> LoadAsync()
...
-    /// Gets the income amount for the date range
+    /// Gets the total income amount for the transactions given
     /// </summary>
-    /// <param name="endDate">The end date</param>
-    /// <param name="startDate">The start date</param>
-    /// <returns>The income amount for the date range</returns>
-    public decimal GetIncome(DateOnly endDate, DateOnly? startDate = null)
+    /// <param name="transactionIds">The ids of transactions to consider</param>
+    /// <returns>The income amount</returns>
+    public decimal GetIncome(IEnumerable<uint>? transactionIds = null)
```
Comment (nlogozzo) [inline NickvisionMoney.Shared/Models/Account.cs:557]: You can simplify these even more to just:
```csharp
public decimal GetIncome(IEnumerable<uint>? transactionIds = null) => (transactionIds ?? Transactions.Keys)
            .Select(id => Transactions[id])
            .Where(transaction => transaction.Type == TransactionType.Income)
            .Sum(transaction => transaction.Amount);
```
and the other methods as well.

**Review by nlogozzo — CHANGES_REQUESTED** [review]

File: NickvisionMoney.Shared/Models/Account.cs (hunk 1 above)
Comment (JoseBritto) [reply NickvisionMoney.Shared/Models/Account.cs]: Done. Thanks

**Comment by JoseBritto** [comment]
I'll refactor the generate graphs to use these rn

**Review by nlogozzo — APPROVED** [review]

**Comment by nlogozzo** [comment]
Thank you so much for this :)

**Pull request merged**